pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -648,7 +646,13 @@ func (s *FlinkStateMachine) handleApplicationDeleting(ctx context.Context, app *
 	return nil
 }
 
-func (s *FlinkStateMachine) getAndUpdateError(ctx context.Context, application *v1alpha1.FlinkApplication, err error) {
+func (s *FlinkStateMachine) compareAndUpdateError(ctx context.Context, application *v1alpha1.FlinkApplication, err error) {
+	oldErr := application.Status.LastSeenError
+
+	if err == nil && oldErr == nil {
+		return
+	}
+
```

Comment:
Can you update `GetRetryableError` to return `FlinkApplicationError` instead of error ?

Reply:
I could. But I'm somewhat leaning towards not doing that; If the `api` returned FlinkApplicationError, which was then passed on to the controller (and then the state machine), we'd have to update all of the function signatures to return FlinkApplicationError(s).